### PR TITLE
mypy: Added mypy typing for`greet_device_widget` and `greet_user_widget`

### LINKED ISSUE #3392: [`core.gui`] Add mypy typing for `greet_*` widgets

### PR DESCRIPTION
**Pull request by Volham22**
This close #3392

### CODE REVIEW

**Review by TimeEngineer — APPROVED** [review]

File: parsec/core/gui/greet_device_widget.py [hunk 1]
```diff
@@ -85,38 +100,40 @@ async def run(self, core, token):
...
 
-    async def wait_peer_trust(self):
+    async def wait_peer_trust(self) -> None:
         await self.main_mc_send.send(self.Step.WaitPeerTrust)
         r, exc = await self.job_mc_recv.receive()
         if not r:
             raise JobResultError(status="wait-peer-trust-failed", origin=exc)
 
-    async def get_claimer_sas(self):
+    async def get_claimer_sas(self) -> Tuple[object, object]:
```
Comment (Max-7) [inline parsec/core/gui/greet_device_widget.py:120]: Should be `Tuple[SASCode, list[SASCode]]`

File: parsec/core/gui/greet_device_widget.py [hunk 2]
```diff
@@ -457,20 +480,20 @@ class GreetDeviceWidget(QWidget, Ui_GreetDeviceWidget):
     greeter_success = pyqtSignal(QtToTrioJob)
     greeter_error = pyqtSignal(QtToTrioJob)
 
-    def __init__(self, core, jobs_ctx, invite_addr):
+    def __init__(self, core: LoggedCore, jobs_ctx: Any, invite_addr: BackendInvitationAddr) -> None:
```
Comment (Max-7) [inline parsec/core/gui/greet_device_widget.py:483]: `jobs_ctx` is a `QtToTrioJobScheduler`

File: parsec/core/gui/greet_device_widget.py [hunk 3]
```diff
@@ -575,11 +608,11 @@ def cancel(self):
         if self.greeter_job:
             self.greeter_job.cancel()
 
-    def on_close(self):
+    def on_close(self) -> None:
         self.cancel()
 
     @classmethod
-    def show_modal(cls, core, jobs_ctx, invite_addr, parent, on_finished):
+    def show_modal(cls, core: LoggedCore, jobs_ctx: Any, invite_addr: BackendInvitationAddr, parent: QWidget, on_finished: Callable[..., None]) -> GreetDeviceWidget:  # type: ignore[misc]
```
Comment (Max-7) [inline parsec/core/gui/greet_device_widget.py:615]: `jobs_ctx` is a `QtToTrioJobScheduler`

File: parsec/core/gui/greet_user_widget.py [hunk 4]
```diff
@@ -118,37 +130,37 @@ async def run(self, core, token):
...
 
-    async def wait_peer_trust(self):
+    async def wait_peer_trust(self) -> None:
         await self.main_mc_send.send(self.Step.WaitPeerTrust)
         r, exc = await self.job_mc_recv.receive()
         if not r:
             raise JobResultError(status="wait-peer-trust-failed", origin=exc)
 
-    async def get_claimer_sas(self):
+    async def get_claimer_sas(self) -> Tuple[object, object]:
```
Comment (Max-7) [inline parsec/core/gui/greet_user_widget.py:150]: `Tuple[SASCode, list[SASCode]]`

File: parsec/core/gui/greet_user_widget.py [hunk 5]
```diff
@@ -172,7 +184,7 @@ class GreetUserInstructionsWidget(QWidget, Ui_GreetUserInstructionsWidget):
     wait_peer_success = pyqtSignal(QtToTrioJob)
     wait_peer_error = pyqtSignal(QtToTrioJob)
 
-    def __init__(self, jobs_ctx, greeter):
+    def __init__(self, jobs_ctx: Any, greeter: Any) -> None:
```
Comment (Max-7) [inline parsec/core/gui/greet_user_widget.py:187]: `QtToTrioJobScheduler` and `Greeter`

File: parsec/core/gui/greet_user_widget.py [hunk 6]
```diff
@@ -229,7 +241,9 @@ class GreetUserCheckInfoWidget(QWidget, Ui_GreetUserCheckInfoWidget):
     create_user_success = pyqtSignal(QtToTrioJob)
     create_user_error = pyqtSignal(QtToTrioJob)
 
-    def __init__(self, jobs_ctx, greeter, user_profile_outsider_allowed=False):
+    def __init__(
+        self, jobs_ctx: Any, greeter: Any, user_profile_outsider_allowed: bool = False
```
Comment (Max-7) [inline parsec/core/gui/greet_user_widget.py:245]: `QtToTrioJobScheduler` and `Greeter`

File: parsec/core/gui/greet_user_widget.py [hunk 7]
```diff
@@ -392,7 +407,7 @@ class GreetUserCodeExchangeWidget(QWidget, Ui_GreetUserCodeExchangeWidget):
     get_greeter_sas_success = pyqtSignal(QtToTrioJob)
     get_greeter_sas_error = pyqtSignal(QtToTrioJob)
 
-    def __init__(self, jobs_ctx, greeter):
+    def __init__(self, jobs_ctx: Any, greeter: Any) -> None:
```
Comment (Max-7) [inline parsec/core/gui/greet_user_widget.py:410]: `QtToTrioJobScheduler` and `Greeter`

File: parsec/core/gui/greet_user_widget.py [hunk 8]
```diff
@@ -575,20 +592,20 @@ class GreetUserWidget(QWidget, Ui_GreetUserWidget):
     greeter_success = pyqtSignal(QtToTrioJob)
     greeter_error = pyqtSignal(QtToTrioJob)
 
-    def __init__(self, core, jobs_ctx, token):
+    def __init__(self, core: LoggedCore, jobs_ctx: Any, token: InvitationToken) -> None:
```
Comment (Max-7) [inline parsec/core/gui/greet_user_widget.py:595]: `QtToTrioJobScheduler`

File: parsec/core/gui/greet_user_widget.py [hunk 9]
```diff
@@ -711,11 +742,11 @@ def cancel(self):
         if self.greeter_job:
             self.greeter_job.cancel()
 
-    def on_close(self):
+    def on_close(self) -> None:
         self.cancel()
 
     @classmethod
-    def show_modal(cls, core, jobs_ctx, token, parent, on_finished):
+    def show_modal(cls, core: LoggedCore, jobs_ctx: Any, token: InvitationToken, parent: QWidget, on_finished: Callable[..., None]) -> GreetUserWidget:  # type: ignore[misc]
```
Comment (Max-7) [inline parsec/core/gui/greet_user_widget.py:749]: `QtToTrioJobScheduler`

**Review by Max-7 — CHANGES_REQUESTED** [review]

File: parsec/core/gui/greet_device_widget.py [hunk 10]
```diff
@@ -137,7 +154,9 @@ class GreetDeviceInstructionsWidget(QWidget, Ui_GreetDeviceInstructionsWidget):
     send_email_success = pyqtSignal(QtToTrioJob)
     send_email_error = pyqtSignal(QtToTrioJob)
 
-    def __init__(self, jobs_ctx, greeter, invite_addr, core):
+    def __init__(
+        self, jobs_ctx: Any, greeter: Any, invite_addr: BackendInvitationAddr, core: LoggedCore
```
Comment (Max-7) [inline parsec/core/gui/greet_device_widget.py:158]: `QtToTrioJobScheduler` and `Greeter` (may have missed it the first time, sorry)

File: parsec/core/gui/greet_device_widget.py [hunk 11]
```diff
@@ -271,7 +292,7 @@ class GreetDeviceCodeExchangeWidget(QWidget, Ui_GreetDeviceCodeExchangeWidget):
     get_greeter_sas_success = pyqtSignal(QtToTrioJob)
     get_greeter_sas_error = pyqtSignal(QtToTrioJob)
 
-    def __init__(self, jobs_ctx, greeter):
+    def __init__(self, jobs_ctx: Any, greeter: Any) -> None:
```
Comment (Max-7) [inline parsec/core/gui/greet_device_widget.py:295]: `QtToTrioJobScheduler` and `Greeter`

**Review by Max-7 — CHANGES_REQUESTED** [review]

**Review by Max-7 — APPROVED** [review]

**Review by vxgmichel — APPROVED** [review]

**Pull request merged**